### PR TITLE
[UXE-2340] feat implements new type origins

### PR DESCRIPTION
**Pull request by lucasmendes21**
### What is the new behavior introduced by this PR?
implements new type origins of type object storange

### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.

https://github.com/aziontech/azion-console-kit/assets/92529166/7c45e2e9-8ad6-43cb-93be-bc4fed754f65

### Does it have a link on Figma?
no

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (fix, feat, test, etc)
- [ ] User inputs are sanitized/protected against malicious attacks
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Review by HerbertJulio — DISMISSED** [review]

File: src/views/EdgeApplicationsOrigins/Drawer/index.vue [hunk 1]
```diff
@@ -85,22 +90,29 @@
...
+      is: (originType) => originType === 'object_storage',
+      then: (schema) => schema.optional(),
+      otherwise: (schema) => schema.of(
+        yup.object().shape({
+          address: yup.string().label('Address').required(),
+          weight: yup.number().nullable().label('Weight'),
+          isActive: yup.boolean().default(true).label('Active')
+        })
+      ),
+    }),
```
Comment (HerbertJulio) [inline src/views/EdgeApplicationsOrigins/Drawer/index.vue:115]: Muito bom 👍

**Review by pauloSF0 — DISMISSED** [review]

**Review by peterpaulo-azion — APPROVED** [review]

**Review by peterpaulo-azion — DISMISSED** [review]

**Review by pauloSF0 — APPROVED** [review]

**Review by HerbertJulio — APPROVED** [review]

File: src/views/EdgeApplicationsOrigins/Drawer/index.vue [hunk 2]
```diff
@@ -60,7 +60,12 @@
       label: 'Load Balancer',
       value: 'load_balancer',
       disabled: !props.isLoadBalancer
-    }
+    },
+    {
+      label: 'Object Storage',
```
Comment (bru-andrade) [inline src/views/EdgeApplicationsOrigins/Drawer/index.vue:65]:
```suggestion
      label: 'Edge Storage',
```

File: src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue [hunk 3]
```diff
@@ -301,6 +304,22 @@
           blank, Azion will use the Address as the default."
         />
       </div>
+      <div class="flex flex-col sm:max-w-lg w-full gap-2" v-if="isObjectStorageOriginType">
+        <FieldText
+          label="Bucket Name *"
+          name="bucketName"
+          :value="bucketName"
+          description="Name of the bucket created using Azion Object Storage."
```
Comment (bru-andrade) [inline src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue:312]:
```suggestion
          description="Name of the bucket created using Azion Edge Storage."
```

File: src/views/EdgeApplicationsOrigins/Drawer/index.vue (hunk 2 above)
Comment (lucasmendes21) [reply src/views/EdgeApplicationsOrigins/Drawer/index.vue:65]: Vai ser chama Edge Storage agora? @bru-andrade

File: src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue (hunk 3 above)
Comment (lucasmendes21) [reply src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue:312]: Vai ser chama Edge Storage agora? @bru-andrade

Comment (bru-andrade) [reply src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue:312]: Aqui precisaria ser alterado por conta do nome do produto.

File: src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue [hunk 4]
```diff
@@ -301,6 +304,22 @@
...
+        />
+      </div>
+      <div class="flex flex-col sm:max-w-lg w-full gap-2" v-if="isObjectStorageOriginType">
+        <FieldText
+          label="Prefix"
+          name="prefix"
+          :value="prefix"
+          description="Path or directory within the bucket that will serve as the origin. Leave blank if you want to use the root directory."
```
Comment (bru-andrade) [inline src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue:320]: Nice 👍🏻

File: src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue (hunk 3 above)
Comment (bru-andrade) [reply src/views/EdgeApplicationsOrigins/FormFields/FormFieldsEdgeApplicationsOrigins.vue:312]: Sim, é o nome oficial, mudou faz pouco tempo 🥴 

https://www.azion.com/en/documentation/products/store/edge-storage/

**Review by bru-andrade — APPROVED** [review]

**Review by william-tome — APPROVED** [review]

**Pull request merged**